modules/storage-page-memory/src/main/java/org/apache/ignite/internal/storage/pagememory/mv/AbstractPageMemoryMvPartitionStorage.java

```diff
@@ -216,9 +217,11 @@ private PageMemoryHashIndexStorage createOrRestoreHashIndex(IndexMeta indexMeta)
             );
 
             if (initNew) {
-                boolean replaced = indexMetaTree.putx(new IndexMeta(indexMeta.id(), metaPageId, inlineSize));
+                boolean replaced = indexMetaTree.putx(new IndexMeta(indexMeta.id(), metaPageId));
 
                 assert !replaced;
+
+                writeHashIndexInlineSize(metaPageId, inlineSize);
```

Comment:
Shouldn't `HashIndexTree` do it? Why is it an external code if it's the part of initialization? Please fix it

Reply:
I'll try to move it inside the tree.

Reply:
Fix it